testing/integration/upgrade_test.go

```diff
@@ -0,0 +1,160 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build integration
+
+package tests
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/Masterminds/semver"
+	"github.com/google/uuid"
+
+	"github.com/elastic/elastic-agent-libs/kibana"
+
+	atesting "github.com/elastic/elastic-agent/pkg/testing"
+	"github.com/elastic/elastic-agent/pkg/testing/define"
+	"github.com/elastic/elastic-agent/pkg/testing/tools"
+	"github.com/elastic/elastic-agent/version"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type UpgradeElasticAgent struct {
+	suite.Suite
+
+	requirementsInfo  *define.Info
+	agentStartVersion string
+	agentEndVersion   string
+	agentFixture      *atesting.Fixture
+}
+
+func newUpgradeElasticAgentTestSuite(info *define.Info, startVersion, endVersion string) *UpgradeElasticAgent {
+	return &UpgradeElasticAgent{
+		requirementsInfo:  info,
+		agentStartVersion: startVersion,
+		agentEndVersion:   endVersion,
+	}
+}
+
+// Before suite
+func (s *UpgradeElasticAgent) SetupSuite() {
+	agentFixture, err := atesting.NewFixture(
+		s.T(),
+		s.agentStartVersion,
+		atesting.WithFetcher(atesting.ArtifactFetcher()),
+	)
+	require.NoError(s.T(), err)
+	s.agentFixture = agentFixture
+}
+
+func (s *UpgradeElasticAgent) TestUpgradeFleetManagedElasticAgent() {
+	policyUUID := uuid.New().String()
+
+	createPolicyReq := kibana.CreatePolicyRequest{
+		Name:        "test-policy-" + policyUUID,
+		Namespace:   "default",
+		Description: "Test policy " + policyUUID,
+		MonitoringEnabled: []kibana.MonitoringEnabledOption{
+			kibana.MonitoringEnabledLogs,
+			kibana.MonitoringEnabledMetrics,
+		},
+	}
+	policy, err := s.requirementsInfo.KibanaClient.CreatePolicy(createPolicyReq)
+	require.NoError(s.T(), err)
+
+	createEnrollmentApiKeyReq := kibana.CreateEnrollmentAPIKeyRequest{
+		PolicyID: policy.ID,
+	}
+	enrollmentToken, err := s.requirementsInfo.KibanaClient.CreateEnrollmentAPIKey(createEnrollmentApiKeyReq)
+	require.NoError(s.T(), err)
+
+	// TODO: figure out how to get Fleet Server URL
+	fleetServerURL := ""
```

Comment:
TODO: we might need an enhancement in https://github.com/elastic/elastic-agent-libs/blob/main/kibana/fleet.go to return the Fleet Server URL.